CMakeLists.txt references moodycamel which has been removed

`moodycamel` was removed in https://github.com/ROCm-Developer-Tools/HIP-CPU/commit/5eeb1b3a3235486824c19a315ace2370ba6b973e but this change was not reflected in `CMakeLists.txt`. We came upon this issue while building rocRAND with HIP CPU:

In `hip-cpu-download/hip-cpu-download-prefix/src/hip-cpu-download-stamp/hip-cpu-download-install-err.log`
```log
CMake Error at cmake_install.cmake:81 (file):
  file INSTALL cannot find
  "/workspaces/amd/libraries/rocRAND/build/hip-cpu-src/external/moodycamel":
  No such file or directory.
```

After removing the relevant section in the CMake script it works again.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,9 +90,6 @@ if (hip_cpu_rt_INSTALL_LIBRARY)
         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/external/half DESTINATION external)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/external/libco DESTINATION external)
-    install(
-        DIRECTORY
-            ${PROJECT_SOURCE_DIR}/external/moodycamel DESTINATION external)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/hip DESTINATION include)
     install(DIRECTORY ${PROJECT_SOURCE_DIR}/src/include DESTINATION src)
 endif ()

